modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/loader/enricher/MimeTypeParametersDeclarationEnricher.java

```diff
@@ -123,6 +159,53 @@ public void visitString(StringType stringType) {
           replaceOutputType(declaration, property, format -> BaseTypeBuilder.create(format).stringType().build());
         }
 
+        @Override
+        public void visitArrayType(ArrayType arrayType) {
+          if (!outputType.isPresent()) {
+            return;
+          }
+
+          Type type = outputType.get();
+          Type itemType = getItemType(type);
+
+          if (itemType == null) {
+            return;
+          }
+
+          Type resultPayloadType = getResultPayloadType(itemType);
```

Comment:
if itemType is only used to resolve the payloadType then calculate it inside the getPayload method